src/main/java/com/atlassian/db/replica/api/reason/RouteDecision.java

```diff
@@ -38,14 +42,31 @@ public Reason getReason() {
         return Optional.ofNullable(cause);
     }
 
+    /**
+     * @return information whether reason was caused by write sql operation.
+     */
+    public boolean isWrite(){
+        return WRITE_OPERATION.equals(reason) ||
```

Comment:
I want to start discussion here.
I think this answer will work in majority of the cases but not all.

I think there is no way to have a class like SQLQuery.isSQLModifyDatabase(sql) - though I have to say It was tempting for me in place to think about it, it would be extracted from `ReplicaStatement`
with methods like
`sqlFunction.isFunctionCall(sql) || isUpdate(sql) || isDelete(sql);`
But why i didn't do that?
Cause this is not enough. I think SQL + statementType + still there are chances that 
`ReplicaPreparedStatement.execute() ` will run either READ or Write operation.

I wonder whether we should introduce localState(localField) in `ReplicaStatement` that would determine whether query isReadOnly or not.
(it might be easier to determine whether query is ReadOnly, or whether isWrite).
@wyrzyk How would you approach it ?

Reply:
P.S. 
Initially I thought about extending `Reason` and adding one more boolean field isWrite.

But some of states might be either write or read or changed. I think initliallty `Reason` was created present other dimensions. On from those dimensions you cannot ultimately judge whether Reason was write or read. 

Am I right ? 
I'm asking for suggestions.

Reply:
We always know if the query is a read or write operation. That's the library's main goal: to split read and write operations between the main database and replicas.

Extending `Reason` with `isWrite` seems to be a good call.

> But some of states might be either write or read or changed

What do you mean? Can you elaborate?

Reply:
@wyrzyk I've created a draft PR especially to elaborate my thinking here: 
Please take a look. I've added a bunch of todo/comments overthere, looking for a best place to decide whether query is read/write, cause for me there are couple of places when actually we don;t know, or for safety reasons we treat querys as writes, while based on SQL they aren't writes.

https://github.com/atlassian-labs/db-replica/pull/49

Reply:
> or for safety reasons we treat querys as writes, while based on SQL they aren't writes.

Yes. That's true. Some of the places are tracked in GH issues. I guess we're going to resolve some of these issues when we gain better observability.

Reply:
At the end in draft it turns out that there are only two types of Writes operations (`RW_API_CALL` , `WRITE_OPERATION`
Rest is read, but might be treated as Write due to circumstances with connection state, transaction lvl,etc.
@dagguh 
@wyrzyk